lib/3scale/client.rb

```diff
@@ -291,8 +307,19 @@ def oauth_authorize(options)
     ALL_PARAMS = [:user_key, :app_id, :app_key, :service_id, :redirect_url, :usage]
     REPORT_PARAMS = [:user_key, :app_id, :service_id, :timestamp]
 
+    def creds_params(options)
```

Comment:
This method is doing 2 things:
1) checking that both `service_token` and `service_id` are present when using service_tokens instead of provider keys.
2) setting `@creds_params`.

Would it help readability splitting the method?